### PR TITLE
intl 0.19, remove greenrobot.eventbus to support latest version of Flutter

### PR DESCRIPTION
**Pull request by atavism**

### CODE REVIEW

File: pubspec.yaml
Comment (jigar-f) [inline pubspec.yaml:1]: @atavism  While we are here, I think we should go ahead and update another plugin as well, I think there are quite a few we can update.

**Review by jigar-f — APPROVED** [review]
LGTM!, Just two comments.

**Comment by jigar-f** [comment]
@atavism The Only issue with upgrading to Flutter is this crash on android. any version less than 14 it crashes. Do you mind testing this In 11,12 or 13 and see if you are getting crash?

**Comment by atavism** [comment]
> @atavism The Only issue with upgrading to Flutter is this crash on android. any version less than 14 it crashes. Do you mind testing this In 11,12 or 13 and see if you are getting crash?

@jigar-f Thanks! I didn't realize upgrading Flutter causes a crash on Android. Taking a look now

File: pubspec.yaml
Comment (atavism) [reply pubspec.yaml:1]: Sounds good

File: pubspec.yaml [hunk 1]
```diff
@@ -109,51 +107,45 @@ dependencies:
...
+  app_links: ^6.1.1
   #Loading
   animated_loading_border: ^0.0.2
   shimmer: ^3.0.0
 
-
-# wakelock ^0.6.2 requires win32 ^2.0.0 or ^3.0.0
-# See https://github.com/creativecreatorormaybenot/wakelock/issues/211
-dependency_overrides:
-  win32: 3.1.4
```
Comment (atavism) [inline pubspec.yaml:144]: I don't think we need this since we aren't using [wakelock](https://pub.dev/packages/wakelock) anymore

**Comment by atavism** [comment]
Hmm, I'm not seeing the app crash on Android with these changes.. What error are you seeing in the logs?

**Comment by atavism** [comment]
Ok, the app is crashing for me testing with a Samsung Galaxy S9 running Android 9:

```
06-20 10:26:53.954  6868  6868 E AndroidRuntime: Process: org.getlantern.lantern, PID: 6868
06-20 10:26:53.954  6868  6868 E AndroidRuntime: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/window/BackEvent;
06-20 10:26:53.954  6868  6868 E AndroidRuntime: 	at java.lang.reflect.Executable.getParameterTypesInternal(Native Method)
06-20 10:26:53.954  6868  6868 E AndroidRuntime: 	at java.lang.reflect.Method.getParameterTypes(Method.java:186)
06-20 10:26:53.954  6868  6868 E AndroidRuntime: 	at org.greenrobot.eventbus.SubscriberMethodFinder.findUsingReflectionInSingleClass(SubscriberMethodFinder.java:173)
06-20 10:26:53.954  6868  6868 E AndroidRuntime: 	at org.greenrobot.eventbus.SubscriberMethodFinder.findUsingInfo(SubscriberMethodFinder.java:88)
06-20 10:26:53.954  6868  6868 E AndroidRuntime: 	at org.greenrobot.eventbus.SubscriberMethodFinder.findSubscriberMethods(SubscriberMethodFinder.java:64)
06-20 10:26:53.954  6868  6868 E AndroidRuntime: 	at org.greenrobot.eventbus.EventBus.register(EventBus.java:150)
06-20 10:26:53.954  6868  6868 E AndroidRuntime: 	at org.getlantern.lantern.MainActivity.onCreate(MainActivity.kt:135)
06-20 10:26:53.954  6868  6868 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:7326)
06-20 10:26:53.954  6868  6868 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:7317)
06-20 10:26:53.954  6868  6868 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1271)
06-20 10:26:53.954  6868  6868 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3072)
06-20 10:26:53.954  6868  6868 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3235)
06-20 10:26:53.954  6868  6868 E AndroidRuntime: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
06-20 10:26:53.954  6868  6868 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
06-20 10:26:53.954  6868  6868 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
06-20 10:26:53.954  6868  6868 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1926)
06-20 10:26:53.954  6868  6868 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
06-20 10:26:53.954  6868  6868 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:214)
06-20 10:26:53.954  6868  6868 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6986)
06-20 10:26:53.954  6868  6868 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
06-20 10:26:53.954  6868  6868 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
06-20 10:26:53.954  6868  6868 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1445)
```

**Comment by atavism** [comment]
> Ok, the app is crashing for me testing with a Samsung Galaxy S9 running Android 9:

It's related to this issue here: https://github.com/pichillilorenzo/flutter_inappwebview/issues/2139

**Comment by atavism** [comment]
Ok, removing the `org.greenrobot.eventbus` package resolves the issue. I am making some changes to handle events using Kotlin flows instead

**Comment by jigar-f** [comment]
thanks, @atavism for looking into this, I will test this shortly.

**Comment by jigar-f** [comment]
@atavism Can we please add these proguard rules as well?
https://github.com/pichillilorenzo/flutter_inappwebview/issues/2139#issuecomment-2179928178

**Comment by atavism** [comment]
> @atavism Can we please add these proguard rules as well?
https://github.com/pichillilorenzo/flutter_inappwebview/issues/2139#issuecomment-2179928178

Done https://github.com/getlantern/lantern-client/pull/1103/commits/c44fb24a8f85c411a1815c09a7074e8b1d968c1f

**Pull request merged**